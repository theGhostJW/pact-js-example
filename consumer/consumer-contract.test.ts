import { Pact } from '@pact-foundation/pact';
import { fetchMovies } from './mockConsumer';
import path from 'path';
import { like, eachLike } from '@pact-foundation/pact/src/dsl/matchers';


const port = 8992;
const provider = new Pact({
  consumer: 'movie-consumer',
  provider: 'movie-provider',
  port,
  log: path.resolve(process.cwd(), 'logs', 'pact.log'),
  dir: path.resolve(process.cwd(), 'pacts'),
  pactfileWriteMode: 'update',
  logLevel: 'info',
});

describe('Movies Service', () => {
  describe('When a request to list all movies is made', () => {
    beforeAll(() =>
      provider.setup().then(() => {
        provider.addInteraction({
          state: "I have a list of movies",
          uponReceiving: 'a request to list all movies',
          withRequest: {
            method: 'GET',
            path: '/movies',
          },
          willRespondWith: {
            status: 200,
            body: eachLike(
              {
                id: 1,
                name: like('Movie 1'),
                year: like(2008),
              },
              { min: 5 }
            ),
          },
        })
      })
    )

    test('should return the correct data', async () => {
      const response = await fetchMovies(provider.mockService.baseUrl);
      expect(response).toMatchSnapshot();
    })


    afterEach(() => provider.verify())
    afterAll(() => provider.finalize())
  });
});
