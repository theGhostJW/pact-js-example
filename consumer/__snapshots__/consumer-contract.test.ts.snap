// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Service When a request to list all movies is made should return the correct data 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Movie 1",
    "year": 2008,
  },
  Object {
    "id": 1,
    "name": "Movie 1",
    "year": 2008,
  },
  Object {
    "id": 1,
    "name": "Movie 1",
    "year": 2008,
  },
  Object {
    "id": 1,
    "name": "Movie 1",
    "year": 2008,
  },
  Object {
    "id": 1,
    "name": "Movie 1",
    "year": 2008,
  },
]
`;
